planner/core/rule_column_pruning.go

```diff
@@ -105,6 +105,13 @@ func (la *LogicalAggregation) PruneColumns(parentUsedCols []*expression.Column)
 		return err
 	}
 
+	if len(parentUsedCols) == 0 && len(used) > 0 {
+		// For sql like `select count(1) from (select count(1) from t)`, no column used.
+		// tikv's response will be empty if the inner aggr is pushed down.
+		// We should force some column to be used.
+		used[0] = true
```

Comment:
This fix looks pretty ad-hoc, IMHO a better approach would be extracting columns from `aggregation(constant)`.

Reply:
You are right -_-. I had the same feeling when I tried to fix this issue. Thanks for your feedback and advice.